db/sql_stmt_cache.c

```diff
@@ -477,8 +481,7 @@ int stmt_cache_get(struct sqlthdstate *thd, struct sqlclntstate *clnt,
             if (find_sql_hint_table(rec->cache_hint, (char **)&rec->sql, &data,
                                     &data_sz) == 0) {
                 rec->status |= CACHE_FOUND_STR;
-                query_data_func(clnt, &data, &data_sz, QUERY_STMT_DATA,
```

Comment:
In case you missed it, this should be `QUERY_HINT_DATA`, instead of `QUERY_STMT_DATA`.